src/coreclr/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -10754,8 +10754,11 @@ void __stdcall ProfilerUnmanagedToManagedTransitionMD(MethodDesc *pMD,
 // These do a lot of work for us, setting up Frames, gathering arg info and resolving generics.
   //*******************************************************************************************
 
-HCIMPL2(EXTERN_C void, ProfileEnter, UINT_PTR clientData, void * platformSpecificHandle)
+HCIMPL2_RAW(EXTERN_C void, ProfileEnter, UINT_PTR clientData, void * platformSpecificHandle)
+GCX_COOP();
```

Comment:
The GC mode for profiler enter/leave callbacks was discussed in https://github.com/dotnet/runtime/issues/68760

It would be best to submit this change as separate PR. I do not think that this fix is complete. The thread is not guaranteed to be setup when this callback is called.

Reply:
Additionally, we should run that through the diagnostics tests quickly and confirm we aren't triggering any asserts/violations. @hoyosjs or I can help with that.

Reply:
This was split into https://github.com/dotnet/runtime/pull/88134. The change should be reverted in this PR.